src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -549,7 +559,14 @@ extern "C" void QCALLTYPE RuntimeMethodHandle_InvokeMethod(
             // save the info into ValueClassInfo
             if (pMT->ContainsGCPointers())
             {
-                pValueClasses = new (_alloca(sizeof(ValueClassInfo))) ValueClassInfo(pStackCopy, pMT, pValueClasses);
+                if (pValueClasses == nullptr)
+                {
+                    pValueClasses = new (_alloca(sizeof(ValueClassInfo))) ValueClassInfo(pStackCopy, pMT, nullptr);
+                }
+                else
+                {
+                    pValueClasses->pNext = new (_alloca(sizeof(ValueClassInfo))) ValueClassInfo(pStackCopy, pMT, nullptr);
+                }
```

Comment:
```suggestion
                pValueClasses = new (_alloca(sizeof(ValueClassInfo))) ValueClassInfo(pStackCopy, pMT, pValueClasses);
```
I do not see why this change is needed. It should not matter whether we append the new entry to the beginning or end of the linked list.

Reply:
Ah, indeed!